packages/shared/src/components/MainLayout.tsx

```diff
@@ -123,8 +188,30 @@ export default function MainLayout({
     setOpenMobileSidebar(state);
   };
 
+  useEffect(() => {
```

Comment:
Are we sure incomplete registration should be a state rather than a derived value from the user object?

Reply:
Right. We can just derive this from the user object without the needing of any state.